modules: mcuboot: Add override to MCUBoot hooks (#10022)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

The following sections provide detailed lists of changes by component.

MCUboot
=======

* Added:

  * An option to prevent the inclusion of the default nRF5340 network core DFU image hook, which allows a custom implementation by users if the :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_NRF5340` Kconfig option is disabled (it is enabled by default).

>     CMake can now be used to add additional hook files.

Reviewer: This comment is not true for this PR

The following sections provide detailed lists of changes by component.

MCUboot
=======

* Added:

>   * An option to prevent inclusion of default nRF5340 network core DFU image hook, which allows a custom implementation by users if the :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_NRF5340` Kconfig option is disabled (enabled by default).
>     CMake can be used to add additional hook files.
>     See :file:`modules/mcuboot/hooks/CMakeLists.txt` for an example of how to achieve this.
>   * Support for using custom boot image hook file through Kconfig option :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_CUSTOM` and :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_CUSTOM_PATH`.

Reviewer 2: ```suggestion
  * An option to prevent the inclusion of the default nRF5340 network core DFU image hook, which allows a custom implementation by users if the :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_NRF5340` Kconfig option is disabled (it is enabled by default).
    CMake can now be used to add additional hook files.
  * Support for using a custom boot image hook file through Kconfig option :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_CUSTOM` and :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOK_CUSTOM_PATH`.
```

MCUboot
=======

> * Added:
> 
>   * Support for using custom boot image hook file through Kconfig option :kconfig:option:`CONFIG_USE_CUSTOM_BOOT_IMAGE_ACCESS_HOOKS` and :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOKS_PATH <full path to file>`.

Reviewer 2: ```suggestion
* Added support for using a custom boot image hook file through the Kconfig options :kconfig:option:`CONFIG_USE_CUSTOM_BOOT_IMAGE_ACCESS_HOOKS` and :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOKS_PATH <full path to file>`.
```
